pkg/generate/code/set_resource.go

```diff
@@ -770,6 +769,206 @@ func SetResourceGetAttributes(
+	primaryKeyOut += fmt.Sprintf("%s}\n", indent)
+
+	primaryIdentifier := ""
+
+	// Attempt to fetch the primary identifier from the configuration
+	opConfig, ok := cfg.Operations[op.Name]
+	if ok {
+		primaryIdentifier = opConfig.PrimaryIdentifierFieldName
+	}
+
+	// Determine the "primary identifier" based on the names of each field
+	if primaryIdentifier == "" {
+		primaryIdentifierLookup := []string{
+			"Name",
+			r.Names.Original + "Name",
+			r.Names.Original + "Id",
+		}
+
+		for _, memberName := range inputShape.MemberNames() {
+			if util.InStrings(memberName, primaryIdentifierLookup) {
+				if primaryIdentifier == "" {
+					primaryIdentifier = memberName
+				} else {
+					panic("Found multiple possible primary identifiers for " +
+						r.Names.Original + ". Set " +
+						"`primary_identifier_field_name` for the " + op.Name +
+						" operation in the generator config.")
+				}
+			}
+		}
+
+		// Still haven't determined the identifier? Panic
+		if primaryIdentifier == "" {
+			panic("Could not find primary identifier for " + r.Names.Original +
+				". Set `primary_identifier_field_name` for the " + op.Name +
+				" operation in the generator config.")
+		}
+	}
+
+	paginatorFieldLookup := []string{
+		"NextToken",
+		"MaxResults",
+	}
+
+	additionalKeyCount := 0
+	for _, memberName := range inputShape.MemberNames() {
+		if r.UnpacksAttributesMap() && memberName == "Attributes" {
+			continue
+		}
+
+		if util.InStrings(memberName, paginatorFieldLookup) {
+			continue
+		}
+
+		memberShapeRef, _ := inputShape.MemberRefs[memberName]
+		memberShape := memberShapeRef.Shape
+
+		// Only strings are currently accepted as valid inputs for
+		// additional key fields
+		if memberShape.Type != "string" {
+			continue
+		}
+
+		if r.IsSecretField(memberName) {
+			// Secrets cannot be used as fields in identifiers
+			continue
+		}
+
+		if r.IsPrimaryARNField(memberName) {
+			// r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
+			arnOut += fmt.Sprintf(
+				"\n%s%s.Status.ACKResourceMetadata.ARN = %s.ARN\n",
+				indent, targetVarName, sourceVarName,
+			)
+			continue
+
+		}
+
+		isPrimaryIdentifier := memberName == primaryIdentifier
+		cleanMemberNames := names.New(memberName)
+		cleanMemberName := cleanMemberNames.Camel
+
+		memberPath := ""
+		_, inSpec := r.SpecFields[memberName]
+		_, inStatus := r.StatusFields[memberName]
+		switch {
+		case inSpec:
+			memberPath = "Spec"
+		case inStatus:
+			memberPath = "Status"
+		case isPrimaryIdentifier:
+			panic("Primary identifier field '" + memberName + "' cannot be found in either spec or status.")
+		default:
+			continue
+		}
+
+		if isPrimaryIdentifier {
+			// r.ko.Status.BrokerID = identifier.NameOrID
+			primaryKeyOut += fmt.Sprintf("%s%s.%s.%s = %s.NameOrID\n", indent, targetVarName, memberPath, cleanMemberName, sourceVarName)
+		} else {
+			// f0, f0ok := identifier.AdditionalKeys["scalableDimension"]
```

Comment:
Did we plan to support case-insensitive lookups? Because if the `Spec` or `Status` field is PascalCased, per Go export rules, and we're asking users to use lowerCamelCased, things might get confusing here.

I think we should probably support case-insensitive key lookups here.

Thoughts?

Reply:
All other fields in the k8s spec are lowerCamelCased, I am trying to make the `additionalKeys` map feel more "k8s native".

Reply:
@RedbackThomson yeah, I guess that's a fair point. I just worry that without the TODO below on line 953 being implemented, that simple typos here will be quite difficult to detect.